### ISSUE TITLE
Greyhole on arch linux

### ISSUE THREAD

**Issue by ajwtech** [gboudreau/Greyhole#205]
Two issues. 

On arch linux samba service is smb not smbd, changing the name in the service unit fixed this. 
Second issue greyhole.so vfs in the samba vfs prevents greyhole from starting. I have deleted it but it is put back when the service starts and and then fails with:

```* greyhole.service - Greyhole
   Loaded: loaded (/usr/lib/systemd/system/greyhole.service; enabled; vendor preset: disabled)
   Active: failed (Result: start-limit-hit) since Fri 2019-03-29 21:24:26 EDT; 3s ago
  Process: 6049 ExecStartPre=/usr/bin/greyhole --test-config (code=exited, status=0/SUCCESS)
  Process: 6083 ExecStart=/usr/bin/greyhole --daemon (code=killed, signal=TERM)
 Main PID: 6083 (code=killed, signal=TERM)

Mar 29 21:24:26 hostnamefastvod greyhole[6049]: Config is OK
Mar 29 21:24:26 hostnamefastvod systemd[1]: Started Greyhole.
Mar 29 21:24:26 hostnamefastvod greyhole[6083]: ln: failed to create symbolic link '/usr/lib/samba/vfs/greyhole.so': File exists
Mar 29 21:24:26 hostnamefastvod systemd[1]: Stopping Greyhole...
Mar 29 21:24:26 hostnamefastvod systemd[1]: greyhole.service: Main process exited, code=killed, status=15/TERM
Mar 29 21:24:26 hostnamefastvod systemd[1]: greyhole.service: Succeeded.
Mar 29 21:24:26 hostnamefastvod systemd[1]: Stopped Greyhole.
Mar 29 21:24:26 hostnamefastvod systemd[1]: greyhole.service: Start request repeated too quickly.
Mar 29 21:24:26 hostnamefastvod systemd[1]: greyhole.service: Failed with result 'start-limit-hit'.
Mar 29 21:24:26 hostnamefastvod systemd[1]: Failed to start Greyhole.
```

**Comment by gboudreau** [comment]
Greyhole generates its own logs; look in `/var/log/greyhole.log` for errors that would prevent the daemon from starting.

Also: https://github.com/gboudreau/Greyhole/wiki/GetHelp#gather-system-information

**Comment by ajwtech** [comment]
Last night:
- I modified the service unit so it no longer restarted on its own. 
- I removed the vfs greyhole.so from the samba directory and started the service 
- I restarted the server. 

I found that the new error I got was too many levels in the symbolic link, this did not show up in the logs. From there I traced back the vfs symbolic link and found that `/usr/lib64/greyhole/greyhole-samba49.so` was a symbolic link pointing to itself. I deleted it and replaced it with a good copy from your repo and restarted the server now the issue is resolved. 
I resolved the issue but I am not sure what is caused it.   The system info is here: http://ix.io/1ERt in case you see something I am missing.
